File: sku/src/neon/sku/Sku.scala
package neon.sku

import neon.common.{SkuId, UomHierarchy}

case class Sku(
    id: SkuId,
    code: String,
    description: String,
    lotManaged: Boolean,
    uomHierarchy: UomHierarchy = UomHierarchy.empty
)

File: web/src/shared/types/sku.ts
import type { SkuId } from "./ids"
import type { PackagingLevel } from "./enums"

export type UomHierarchy = Partial<Record<PackagingLevel, number>>

export interface Sku {
  id: SkuId
  code: string
  description: string
  lotManaged: boolean
  uomHierarchy: UomHierarchy
}
